Jumpit/JumpitAPI.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  JumpitAPI.swift
+//  Jumpit
+//
+//  Created by 이지훈 on 5/22/24.
+//
+
+import Foundation
+import Moya
+
+enum JumpitAPI {
+    case postResume(data: ResumeData)
+    case fetchMyResume(userId: String)
+    case toggleResumePrivacy(resumeId: String)
+    case searchPositions(keyword: String)
+    case searchPositionsFiltered(keyword: String, categories: [Int])
+    case fetchPositionDetail(positionId: String)
+}
+
+extension JumpitAPI: TargetType {
+    var baseURL: URL {
+        guard let urlString = Bundle.main.object(forInfoDictionaryKey: "BaseURL") as? String,
+              let url = URL(string: urlString) else {
+            fatalError("BaseURL not set in plist for this environment")
+        }
+        return url
+    }
+
+    var path: String {
+        switch self {
+        case .postResume:
+            return "/resumes"
+        case .fetchMyResume(let userId):
+            return "/resumes/\(userId)"
+        case .toggleResumePrivacy(let resumeId):
+            return "/resumes/\(resumeId)"
+        case .searchPositions, .searchPositionsFiltered:
+            return "/positions"
+        case .fetchPositionDetail(let positionId):
+            return "/positions/\(positionId)"
+        }
+    }
+
```

Comment:
```suggestion
extension JumpitAPI: JumpitBaseAPI {
    var domain: JumpitAPIDomain {
            return .resumes
        }
    }

    var path: String {
        switch self {
        case .postResume:
            return ""
        case .fetchMyResume(let userId):
            return "/\(userId)"
        case .toggleResumePrivacy(let resumeId):
            return "/\(resumeId)"
        }
    }

```
<1번 방식> JumpitAPIDomain에서 분기 처리를 해줬다면 이런식으로 JumpitAPIDomain을 사용해서 나눠 적는 것이 좋을 것 같습니다! 이렇게 함께 적어주고 baseurl을 다시 적어주면 JumpitBaseApi를 쓰는 이유가 모호할것 같습니다!